src/filters/filter_lua_script.cpp

```diff
@@ -0,0 +1,382 @@
+        }
+        for (size_t i = 0; i < names_in_float.size(); i++) {
+            if(!input_channels.float_channels.contains(names_in_float.at(i))) {
+                throw filter_config_exception("Unable to link input of lua filter: channel mapping does not contain channel '" + names_in_float.at(i) + "' of type 'double'.");
+            }
+            in_float.at(i) = input_channels.float_channels.at(names_in_float.at(i));
+        }
+        for (size_t i = 0; i < names_in_color.size(); i++) {
+            if(!input_channels.color_channels.contains(names_in_color.at(i))) {
+                throw filter_config_exception("Unable to link input of lua filter: channel mapping does not contain channel '" + names_in_color.at(i) + "' of type 'pixel'.");
+            }
+            in_color.at(i) = input_channels.color_channels.at(names_in_color.at(i));
+            // Todo delete in_color;
+            lua[names_in_color.at(i)] = std::ref(input_channels.color_channels.at(names_in_color.at(i)));
+        }
+
+        if (!initial_parameters.contains("script")) {
+            throw filter_config_exception("lua filter: unable to setup the script");
+        }
+
+//        if (!initial_parameters.contains("patching")) {
+//            throw filter_config_exception("lua filter: unable to setup the patching");
+//        }
+
+        // initial_parameters.at("patching") divide
+        // for each fixture
+        //      fix_univ
+        //      fix_name
+        //      fix_ch_size
+        //      fixture local = fixture(fix_univ, fix_name, fix_ch_size);
+        //      ? dynamisch verschieden viele Argumente übergeben?
+//              lua.new_usertype<fixture>("Fixture", "pan", &fixture.channel_values.at(0), "s", &dmxfish::dmx::pixel::saturation, "i", &dmxfish::dmx::pixel::iluminance);
+        //      fixtures.append(local);
+
+
+
+        lua.open_libraries(sol::lib::base, sol::lib::package);
+        lua.open_libraries(sol::lib::math, sol::lib::table, sol::lib::string);
+        lua.set_function("update", []() {
+        });
+        lua.set_function("scene_activated", []() {
+        });
+
+//        for(fixture fix: fixtures){
+//            lua.create_named_table(fix.name, "universe", fix.universe,
+//                                   "first_channel", fix.first_channel);
+//        }
+        lua.create_named_table("output");
+
+        lua.set_function( "hsi_to_rgb", sol::overload(
+                hsi_to_rgb_color,
+                hsi_to_rgb_table
+        ) );
+
+        lua.set_function( "hsi_to_rgbw", sol::overload(
+                hsi_to_rgbw_color,
+                hsi_to_rgbw_table
+        ) );
+
+        try {
+            lua.script(initial_parameters.at("script"));
+        } catch (std::exception &e){
+            throw filter_config_exception(std::string("setup the filter threw an error: ") + e.what());
+        }
+
+        sol::object scene_activated_obj = lua["scene_activated"];
```

Comment:
What happens if the User does not provide this method? Will the filter crash?

Reply:
Yes or no: if script is not privided the filter will throw an error. if scene_activated is not available it will throw another error.
If the user does not do anything with scene_activated, then the function of line 213 is used.

Reply:
OK, we should make sure then, that this error will not crash fish and is being quite informative (as described with issue 78 of the editor).

Reply:
Do we have a test for this behavior now?

Reply:
If not: I'll write one real quick.

Reply:
Never mind, the test is there. I'll merge now.

Reply:
added a little more information to the message